app/controllers/merchants_controller.rb

```diff
@@ -33,6 +33,16 @@ def destroy
 
   def show
     @merchant = Merchant.find_by(id: params[:id])
-    head :not_found unless @merchant
+
+    unless @merchant
+      head :not_found
+      return
+    end
+
+    unless session[:merchant_id] == @merchant.id
+      flash[:status] = :error
+      flash[:message] = "You must log in first."
```

Comment:
The message would probably be something like: "You cannot view this merchant's page" since the a user could technically be logged in. 
We'll need to eventually add a hidden field or conditional to wherever the "My Dashboard" link/button will be located so that it will display to a merchant and not to a guest

Reply:
Changed the message.

A "My Dashboard" button sounds like a button that would go inside the nav bar.  Is that right?

Reply:
yeah I think so